### PR TITLE
Dependencies part 4: remove purrr, stringr

### LINKED ISSUE #176: ENH: minimize dependencies
We've accrued some dependencies over the course of development that aren't critical. Currently, we have 93 total recursive dependencies. That's too many for an infrastructure package, IMO. Civis employees can see a full dependency report [here](https://platform.civisanalytics.com/spa/#/reports/69004).

Minimizing the dependencies of the package will make it easier to install for our users by requiring a much smaller subset of CRAN. It will also enable creating smaller/leaner production R images for applications beyond the images we've used so far.

The current dependency list is:
```
Imports:
    dplyr (>= 0.7.0),
    dbplyr (>= 1.0.0),
    devtools,
    feather,
    future (>= 1.8.0),
    ggplot2,
    httr,
    jsonlite,
    lubridate,
    methods,
    memoise,
    purrr,
    roxygen2,
    stats,
    stringr,
    testthat,
    utils
Suggests:
    knitr,
    rmarkdown,
    mockery,
    R.utils,
    rstudioapi,
    yaml
```

Proposed list:

```
Imports:
    future (>= 1.8.0),
    httr,
    jsonlite,
    methods,
    memoise,
    stats,
    utils
Suggests:
   ggplot2, (civisML plots) 
   feather, (civisML)
   knitr,
   mockery,
   rmarkdown,
   roxygen2,
   R.utils,
   rstudioapi,
   testthat,
   yaml
```
Removals: `devtools, dplyr,DBI, dbplyr, lubridate, purrr, stringr`. I'm not sure about `stringr`, but it would be nice to remove. 

There are some downstream risk in modifying common environments. In particular, removing dependencies from `civis` modifies the package environment of `datascience-r`. However, stable environment management should be handled in that repo, not this one. For example, `datascience-r` should handle the issue of moving `feather` and `future` to `Suggests` rather than `Imports`.

After this change, the only system dependency will be on `xml2`, which `httr` uses to parse HTML. 

The impact on existing users should be low. Whenever they update R to a new minor version, they may have to manually install e.g. `feather`. 

- [x] Move optional features to suggests
- [x] Remove DBI interface
- [x] Remove lubridate, purrr, stringr
- [x] Remove devtools, move roxygen2 to suggests

### PR DESCRIPTION
**Pull request by patr1ckm**
This removes `purrr` and `stringr` from dependencies. 

There is one minor breaking change that should have minimal impact: `fetch_until` (in pagination) now only accepts a function rather than a function and a formula. The original functionality was the only non-trivial functionality provided by either dependency that is not easily replicated in base.

I made an attempt at turning formulas into functions, but it's complicated, and would just be tech debt.  I don't think this is essential functionality for the package, certainly not essential enough to have `rlang` as a dependency.

I think the breaking change will be minimal in scope because I have never received a question about using the pagination functions in 2 years.

The final dependency total is 22, mainly from `httr` and `future`. There are several system dependencies from `httr`. `civis` can now be installed easily in a `rocker/r-ver` (just a base R image) and only takes 691mb compared to the current version on master, which takes ~1gb, a 30% size reduction.

Fixes #176 .

### CODE REVIEW

File: R/civis_ml_plot.R [hunk 1]
```diff
@@ -15,8 +15,8 @@ hist.civis_ml <- function(x, name = NULL, ...) {
     names <- get_model_data(x, "target_columns")
   }
   if (is.array(score_array)) {
-    # 3d array to list of 2d arrays
-    score_array <- purrr::array_tree(score_array, margin = 1)
+    score_array <- lapply(apply(score_array, 1, list),
```
Comment (patr1ckm) [inline R/civis_ml_plot.R]: This solution is appropriate for turning the 3d arrays provided by civisML into a list of 2d arrays.

File: R/pagination_helpers.R [hunk 2]
```diff
@@ -50,7 +50,7 @@ fetch_all <- function(fn, ...) {
 #' This is useful when searching for a particular value or record.
 #'
 #' @param fn The API function to be called.
-#' @param .until A function or formula which returns a boolean value.
+#' @param .until A function which returns a boolean value.
```
Comment (patr1ckm) [inline R/pagination_helpers.R]: This is the breaking change, since the function was exported

File: R/utils.R [hunk 3]
```diff
@@ -116,14 +116,9 @@ camel_to_snake <- function(s) {
   first_cap <- '(.)([A-Z][a-z]+)'
   all_cap <- '([a-z0-9])([A-Z])'
 
-  s <- stringr::str_replace_all(s, first_cap, "\\1_\\2")
-  s <- stringr::str_replace_all(s, all_cap, "\\1_\\2")
-  stringr::str_to_lower(s)
-}
-
-snake_to_camel <- function(s) {
```
Comment (patr1ckm) [inline R/utils.R]: This is unused and unexported

File: R/civis_future.R [hunk 4]
```diff
@@ -181,9 +181,9 @@ fetch_logs.CivisFuture <- function(object, ...){
 
 make_docker_cmd <- function(task_file_id, run_script_file_id) {
   # this loads the same initial packages in the same order as default R.
-  cmd <- "Rscript -e \"civis::download_civis(${run_script_file_id}, 'run.R')\" && \
-    Rscript --default-packages=methods,datasets,utils,grDevices,graphics,stats run.R ${task_file_id}"
-  stringr::str_interp(cmd)
+  sprintf("Rscript -e \"civis::download_civis(%i, 'run.R')\" && \
+    Rscript --default-packages=methods,datasets,utils,grDevices,graphics,stats run.R %i",
+          run_script_file_id, task_file_id)
```
Comment (keithing) [inline R/civis_future.R]: all about that base

**Comment by patr1ckm** [comment]
Agreed!

**Pull request merged**